cleanlab/rank.py

```diff
@@ -541,3 +542,49 @@ def get_confidence_weighted_entropy_for_each_label(
     label_quality_scores = np.log(label_quality_scores + 1) / clipped_scores
 
     return label_quality_scores
+
+
+def get_knn_distance_ood_scores(
+    query_features: np.array, nbrs: NearestNeighbors, k: int
+) -> np.array:
+    """Returns the KNN distance out-of-distribution (OOD) score for each datapoint.
+
+    This is a function to compute OOD scores where higher scores indicate the datapoint is more likely to be OOD.
+
+    Parameters
+    ----------
+    query_features : np.array
+      Feature matrix of shape ``(N, ...)``, where N is the number of datapoints.
+      This is the "query set" of features for each datapoint which are used for nearest neighbor search.
+
+    nbrs : sklearn.neighbors.NearestNeighbors
+      Instantiated NearestNeighbors class object that's been fitted on a dataset in the same feature space.
+      Note that the distance metric and n_neighbors is specified when instantiating this class.
+      See: https://scikit-learn.org/stable/modules/neighbors.html
```

Comment:
Is it possible to write the code such that annoy nbrs object could also be plugged in here? or would that be nontrivial because their APIs are too different?

Reply:
That would be nontrivial since their APIs are too different. 

However, if we can wrap Annoy in a class that behaves like sklearn NearestNeighbors, then we can pass it as an arg.

I added an example in the PR description of the workflow with Annoy.